src/NJsonSchema.CodeGeneration.CSharp/Models/PropertyModel.cs

```diff
@@ -86,15 +93,18 @@ public string JsonIgnoreCondition
 
                 bool IsValueType()
                 {
-                    var valueTypeFormats = new string[] { "date", "date-time", "uuid" };
-                    return _property.ActualTypeSchema.Type switch
+                    var type = _property.ActualTypeSchema.Type;
+                    if (type.IsBoolean() || type.IsInteger() || type.IsNumber())
                     {
-                        JsonObjectType.Boolean => true,
-                        JsonObjectType.Integer => true,
-                        JsonObjectType.Number => true,
-                        JsonObjectType.String => valueTypeFormats.Contains(_property.Format),
-                        _ => false,
+                        return true;
+                    }
+
+                    if (type.IsString())
+                    {
+                        return ValueTypeFormats.Contains(_property.Format);
```

Comment:
Still not very happy with this. For instance, one could override the default type for `Date` in the _CSharpGeneratorSettings_. That means that a custom type could be used that is NOT a value type...

For now I really don't see a perfect solution to this problem. Other than maybe providing a _CSharpGeneratorSettings_ setting to specify the _ValueTypeFormats_ (which could default to the ones used now).

Reply:
I don't really see a good way around this either.. Adding another setting that will have to be maintained in the future is something @RicoSuter normally does not want to do.
Perhaps changing WhenWritingNull to WhenWiritngDefault would work better?